commune.lic

```diff
@@ -0,0 +1,412 @@
+    case bput("sprinkle my #{item} on #{target}", 'You sprinkle', 'Sprinkle what', 'What were you referring to')
+	when 'You sprinkle'
+	  return true
+	else
+	  return false
+	end
+  end
+  
+  def wave_incense(target=@self)
+    empty_cleric_hands
+	case bput("get my #{@flint_lighter}", 'You get', 'You are already holding', 'I could not find', 'What were you referring to')
+	when 'You get' || 'You are already holding'
+	  get_cleric_item('incense')
+      while bput('light my incense with my flint', 'nothing happens', 'bursts into flames', 'much too dark in here to do that') == 'nothing happens'
+        waitrt?
+      end
+	  bput("wave my incense at #{target}", 'You wave')
+	  bput('snuff my incense', 'You snuff out') if holding?('incense')
+    else
+	  missing_item("#{@flint_lighter}")
+	end
+	empty_cleric_hands
+  end
+  
+
+  ######################## DEVOTION ########################
+	
+	def check_devotion()
+	  case bput('commune', @devotion_levels)
+	  when 'After a moment, you feel a clear presence like a warm blanket covering you'
+	    @devotion_level = 16
+	  when 'After a moment, you see a crystal-clear vision of your god who speaks slowly and deliberately'
+	    @devotion_level = 15
+      when 'After a moment, you see a vision of your god who calls to you by name, "My child, though you may'
+	    @devotion_level = 14
+      when 'After a moment, you see a vision of your god who calls to you by name, "Come here, my child'
+	    @devotion_level = 13
+      when 'After a moment, you see a silent vision of your god'
+	    @devotion_level = 12
+      when 'After a moment, you sense a slight pressure on your shoulder'
+	    @devotion_level = 11
+      when 'After a moment, you see a vision of your god, though the visage is cloudy'
+	    @devotion_level = 10
+      when 'After a moment, you sense that your god is pleased with your devotion'
+	    @devotion_level = 9
+      when 'After a moment, you sense that your god knows your name'
+	    @devotion_level = 8
+      when 'After a moment, you sense that your god is aware of your devotion'
+	    @devotion_level = 7
+      when 'After a moment, you sense a distinct link between you and your god'
+	    @devotion_level = 6
+      when 'After a moment, you sense that your efforts have not gone unnoticed'
+	    @devotion_level = 5
+      when 'After a moment, you sense that your god is barely aware of you'
+	    @devotion_level = 4
+      when 'You call out to your god, but there is no answer'
+	    @devotion_level = 3
+      when 'You close your eyes and start to concentrate'
+	    @devotion_level = 2
+      when 'You feel unclean and unworthy'
+	    @devotion_level = 1
+	  when 'You sense nothing special from your communing'
+	    @devotion_level = 0
+	  end
+	  return @devotion_level
+	end
+
+  ##################### CARVING ########################
+  
+  def create_bead(bead_type='meraud')
+    symbols = { 'kertigen' => 'raven', 'divyaush' => 'welkin', 'zachriedek' => 'raccoon',
```

Comment:
I think you could move this to a higher scope and use it in the definition of the `target` arg (if you wanted to).